connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java

```diff
@@ -0,0 +1,316 @@
+import com.telenor.connect.id.ConnectTokensTO;
+import com.telenor.connect.id.TokenStore;
+import com.telenor.connect.id.UserInfo;
+import com.telenor.connect.utils.ConnectUrlHelper;
+import com.telenor.connect.utils.RestHelper;
+import com.telenor.mobileconnect.id.MobileConnectAPI;
+import com.telenor.mobileconnect.operatordiscovery.OperatorDiscoveryAPI;
+import com.telenor.mobileconnect.operatordiscovery.OperatorDiscoveryConfig;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import retrofit.Callback;
+import retrofit.ResponseCallback;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+public class MobileConnectSdkProfile extends AbstractSdkProfile {
+
+    private OperatorDiscoveryConfig operatorDiscoveryConfig;
+    private volatile OperatorDiscoveryAPI.OperatorDiscoveryResult operatorDiscoveryResult;
+    private volatile OperatorDiscoveryAPI operatorDiscoveryApi;
+
+    public MobileConnectSdkProfile(
+            Context context,
+            final OperatorDiscoveryConfig operatorDiscoveryConfig,
+            boolean useStaging,
+            boolean confidentialClient) {
+        super(context, useStaging, confidentialClient);
+        this.operatorDiscoveryConfig = operatorDiscoveryConfig;
+    }
+
+    @Override
+    public HttpUrl getApiUrl() {
+        String host = operatorDiscoveryResult.getMobileConnectApiUrl().host();
+        if (useStaging) {
+            // will have no effect on non-Telenor subscribers
+            host = host.replace(
+                    "connect.telenordigital.com",
+                    "connect.staging.telenordigital.com");
+        }
+        HttpUrl.Builder builder = new HttpUrl.Builder();
+        builder
+                .scheme(operatorDiscoveryResult.getMobileConnectApiUrl().scheme())
+                .host(host);
+        for (String seg : operatorDiscoveryResult.getMobileConnectApiUrl().pathSegments()) {
+            builder.addPathSegment(seg);
+        }
+        return builder.build();
+    }
+
+    @Override
+    public String getClientId() {
+        return operatorDiscoveryResult.getClientId();
+    }
+
+    @Override
+    public String getClientSecret() {
+        return operatorDiscoveryResult.getClientSecret();
+    }
+
+    @Override
+    public String getRedirectUri() {
+        return operatorDiscoveryConfig.getOperatorDiscoveryRedirectUri();
+    }
+
+    @Override
+    public String getExpectedIssuer(String actualIssuer) {
+        // discrepancy between .well-known configuration and the actual issuer returned
+        if (operatorDiscoveryResult.getBasePath().contains("telenordigital.com")) {
+            return actualIssuer;
+        }
+        if (getWellKnownConfig() != null) {
+            return getWellKnownConfig().getIssuer();
+        }
+        return actualIssuer;
+    }
+
+    @Override
+    public List<String> getExpectedAudiences(List<String> actualAudiences) {
+        return new ArrayList<>(actualAudiences);
```

Comment:
For CONNECT ID, it works just like you say. For Mobile Connect, however, I have another discrepancy here.... our own server returns `{operator};{clientId}` (like "telenor-norway;weoirasdhfkjsd"). I have checked GSMA documentation, and I haven't found a recommendation what expected audience should be. Hence the code.